pkg/datapath/loader/base.go

```diff
@@ -293,6 +332,41 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 	log.Infof("Setting up base BPF datapath (BPF %s instruction set, %s clock source)",
 		args[initBPFCPU], clockSource[option.Config.ClockSource])
 
+	// EKS expects the aws daemonset to set up the following rules that
+	// relate to NodePort traffic between nodes:
+	//
+	// # sysctl -w net.ipv4.conf.eth0.rp_filter=2
+	// # iptables -t mangle -A PREROUTING -i eth0 -m comment --comment "AWS, primary ENI" -m addrtype --dst-type LOCAL --limit-iface-in -j CONNMARK --set-xmark 0x80/0x80
+	// # iptables -t mangle -A PREROUTING -i eni+ -m comment --comment "AWS, primary ENI" -j CONNMARK --restore-mark --nfmask 0x80 --ctmask 0x80
+	// # ip rule add fwmark 0x80/0x80 lookup main
+	//
+	// It marks packets coming from another node through eth0, and restores
+	// the mark on the return path to force a lookup into the main routing
+	// table. Without these rules, the "ip rules" set by the cilium-cni
+	// plugin tell the host to lookup into the table related to the VPC for
+	// which the CIDR used by the endpoint has been configured.
+	//
+	// We want to reproduce equivalent rules to ensure correct routing.
+	// See also iptables.addCiliumENIRules().
+	if option.Config.Masquerade && option.Config.IPAM == ipamOption.IPAMENI {
```

Comment:
nit: If there's some common set of configurations across multiple packages, it might be worth introducing a dedicated option function to encapsulate the common checks, maybe something like `option.Config.MultiDevice()` or `MultiHoming()` or similar?

Reply:
Good idea, let's sort out the exact conditions we need first.

Reply:
Turns out we don't want checks on masquerading, so we end up with a simple condition already. Marking as resolved.